### ISSUE TITLE
[Incompatibility - v1.1.2] Assets from GeneralImprovements prevented load by HackPad

### ISSUE THREAD

**Issue by SeedyMcSeedface** [Shaosil/LethalCompanyMods-GeneralImprovements#86]
I have also submitted a [similar report](https://github.com/willis81808/LethalCompany-PortableMultiTool/issues/3) to HackPad's github page, however I don't know if I'll receive a response there as the mod was last updated a month or more ago.

After running through mods one-by-one, I've identified what appears to be an incompatibility of some kind with the [HackPad mod](https://thunderstore.io/c/lethal-company/p/willis81808/HackPad/).

When HackPad is enabled, GeneralImprovments is unable to load it's assets file due to the error "another AssetBundle with the same files is already loaded".
This is the only incompatibility I've found, and this persists even with a profile that is just GeneralImprovements and HackPad with their relative dependencies.

I have provided the profile code for the profile I used for testing along with the LogOutput file.

r2modman profile code: 018d53be-f22b-8c48-32e7-301016c75daa
[LogOutput.log](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/files/14080086/LogOutput.log)

**Comment by Shaosil** [comment]
Ah I didn't realize it worked that way or I wouldn't have named the asset bundle something so simple ha, I'll rename it in the next update.

**Comment by Shaosil** [comment]
This should be fixed in v.1.1.3, going live soon

**Comment by SeedyMcSeedface** [comment]
It unfortunately seems to still be occurring.

[LogOutput.log](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/files/14090073/LogOutput.log)

**Comment by Shaosil** [comment]
Hmm it shouldn't be trying to load the old assets file anymore, but in case something else is, can you try manually deleting the file called assets in the same directory?

**Comment by SeedyMcSeedface** [comment]
That's the weird thing
HackPad has no assets file, unless it's part of the dll

![image](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/assets/155770932/5e3413a7-d6e6-4ab5-8439-416a0249bc85)

**Comment by Shaosil** [comment]
Yeah but GeneralImprovements did until v1.1.3, where I renamed it. I have no idea why HackPad would be confused about it offhand but my theory is if my old file even exists, something may be throwing it off

**Comment by SeedyMcSeedface** [comment]
Okay yeah, I misunderstood what you meant at first.
But I can confirm that there is no assets file in the GeneralImprovements folder.
The folder is automatically cleared before updates by r2modman.

![image](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/assets/155770932/cb7d7676-02f8-40d4-822a-04e31b4b5d41)
[LogOutput.log](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/files/14090390/LogOutput.log)

**Comment by Shaosil** [comment]
Hmmm ok, I'll have to test it later my second and see if I can narrow it down. The error message is acting like it's trying to load it twice for some reason. That's almost certainly not a bug with my mod but I'd like to find out why it's happening at least.

**Comment by SeedyMcSeedface** [comment]
Best of luck - hope it's able to be figured out whenever you're able to.
If there's anything I can do to help let me know.

**Comment by Shaosil** [comment]
> It unfortunately seems to still be occurring.
> 
> [LogOutput.log](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/files/14090073/LogOutput.log)

I found the issue, sort of. I'm not sure if it's a bug or what but Unity seems to be detecting mine as having duplicate files (I'm not sure what checks they are doing to determine that) as another mod's bundle that is already loaded.

I changed the way my asset bundle is being built and specify a CRC hash in the load, and that seems to satisfy it. I tested with your profile code and it loads successfully.

I will update here when the next version is going live - hopefully within a day.
